app/src/main/java/io/apicurio/registry/util/ArtifactTypeUtil.java

```diff
@@ -84,20 +113,47 @@ public static ArtifactType discoverType(ContentHandle content, String contentTyp
         } catch (Exception e) {
             // Apparently it's not JSON.
         }
-        
+
         // Try protobuf (only if we haven't already)
         if (!triedProto) {
             ArtifactType type = tryProto(content);
             if (type != null) {
                 return type;
             }
         }
-        
+
         // Try GraphQL (SDL)
         if (tryGraphQL(content)) {
             return ArtifactType.GRAPHQL;
         }
-        
+
+        // Try the various XML formatted types
+        try (InputStream stream = content.stream()) {
+            Document xmlDocument = threadLocaldocBuilder.get().parse(stream);
+            Element root = xmlDocument.getDocumentElement();
+            String ns = root.getNamespaceURI();
+
+            if (ns == null) {
+                // XML
+                return ArtifactType.XML;
+            }
```

Comment:
This should be the "else" clause in this logic.  Should be:  if isXsd() else if isWsdl() else { type = XML }

Reply:
Note that you can have a XML file with a namespace that is not null.  This is very common.  For example:

```xml
<ns:foo xmlns:ns="uri:my-custom-ns">
</ns:foo>
```

That is a simple XML file where the root element has a namespace of `uri:my-custom-ns`.

Reply:
Yep I was catching that with the last `return XML` - better to do it this way

Reply:
Ah yes OK I missed that.